Cargo.toml

```diff
@@ -3,3 +3,4 @@ members = [
   "tower-http",
   "examples/*",
 ]
+resolver = "2"
```

Comment:
> this PR should be rebased

> Probably, but the bigger issue is first to ensure maintainers want to merge this in to begin with

at least because [resolver = "2" has been added to Cargo.toml meanwhile](https://github.com/tower-rs/tower-http/blob/main/Cargo.toml#L2)

so it would be easier to review for David when he will come here... if he comes here, indeed